pyproject.toml

```
@@ -87,6 +91,9 @@ Examples = "https://github.com/scikit-build/scikit-build-core/tree/main/tests/pa
 cmake_extensions = "scikit_build_core.setuptools.extension:cmake_extensions"
 cmake_source_dir = "scikit_build_core.setuptools.extension:cmake_source_dir"
 
+[project.entry-points."scikit_build.metadata"]
+setuptools_scm = "scikit_build_core.settings.metadata:setuptools_scm_version"
```

```suggestion
setuptools-scm = "scikit_build_core.settings.metadata:setuptools_scm_version"
```

Aren't these supposed to use dashes, according the the linked page?

---

Yep, both are allowed, but I am happy to switch paradigm and use dashes instead. Hopefully the `setuptools_scm` will morph into `vcs-versioning` but there is still the question of the dash/underscore issue.

---

This one would still require `scikit-build-core` to be built in a git repository right? That would make it difficult for package managers because they tend to use tar balls with no `.git`. Is it possible to avoid the dynamic versioning by putting a static file with the version number somewhere (from the package manager script)?

---

This is declaring that scikit-build-core provides an entrypoint called setuptools_scm which can be used to fill in dynamic metadata in the projects that it builds. It is not used by scikit-build-core itself, but projects that are using scikit-build-core as a build backend can specify that it should call the entrypoint on their repos during the build, to achieve dynamic versioning. So only users whose packages are in git repos will specify that entrypoint should be used. It might also be sensible for the entrypoint to look at the `config_settings` for some kind of version override which could be passed in via the build front end, and it would also be straightforward to write another entrypoint which would read a version from a file (if that file wasn't just pyproject.toml) for projects which wanted that option.

---

I still am thinking about moving scikit-build-core over to (well, back to) manual versioning & maybe even going to Flit, just to keep the build process as simple as possible. That wouldn't affect this, which only enabling users of scikit-build-core to use dynamic metadata.